### PR TITLE
[PORT FROM HORIZON] [MDB IGNORE] OVERMAP + TRADERS + PLANETS + MINING + EXCAVATION [CREDITS TO AZARAK] [READY FOR TESTING]

### PR DESCRIPTION
**Pull request by Gandalf2k15**
# Everything below was created by Azarak. And you should show him your love. Seriously.

## About The Pull Request

This is a major update and will require test-merging before merge.

Due to the fact these are all interlinked, the option of only porting one was not available.

## What this actually adds
It adds a fucking lot. See below PR's.

This is a port of the following: 
https://github.com/hrzntal/horizon/pull/198
https://github.com/hrzntal/horizon/pull/200
https://github.com/hrzntal/horizon/pull/213
https://github.com/hrzntal/horizon/pull/219
https://github.com/hrzntal/horizon/pull/220
https://github.com/hrzntal/horizon/pull/223
https://github.com/hrzntal/horizon/pull/233
https://github.com/hrzntal/horizon/pull/238
https://github.com/hrzntal/horizon/pull/240
https://github.com/hrzntal/horizon/pull/241
https://github.com/hrzntal/horizon/pull/244
https://github.com/hrzntal/horizon/pull/247
https://github.com/hrzntal/horizon/pull/248
https://github.com/hrzntal/horizon/pull/258
https://github.com/hrzntal/horizon/pull/268
https://github.com/hrzntal/horizon/pull/276
https://github.com/hrzntal/horizon/pull/273
https://github.com/hrzntal/horizon/pull/312
https://github.com/hrzntal/horizon/pull/317
https://github.com/hrzntal/horizon/pull/331
https://github.com/hrzntal/horizon/pull/358
https://github.com/hrzntal/horizon/pull/367
https://github.com/hrzntal/horizon/pull/404
https://github.com/hrzntal/horizon/pull/445
https://github.com/hrzntal/horizon/pull/458
https://github.com/hrzntal/horizon/pull/477
https://github.com/hrzntal/horizon/pull/502

This is a work in progress. It should be completed within the week. Again, all credit goes to Azarak for everything in this PR.

Please show your love and appreciation in the above PR's.

# Gallery
dreamseeker_FGnQJeq5Lz
dreamseeker_3FUOU4iCiG
dreamseeker_6x65xtbCLy
dreamseeker_HvTWQrtIot
dreamseeker_POIDz2JaeX
dreamseeker_ZXb1mGwcOi
dreamseeker_MJUbbnH7Ht

## Changelog
:cl: Azarak
add: An overmap system has been added. [INFO PR HERE]  Port from Horizon. Credits to Azarak.
add: A trader system has been added.  [INFO PR HERE] Port from Horizon. Credits to Azarak.
add: A planet system has been added.  [INFO PR HERE] Port from Horizon. Credits to Azarak.
add: An excavation system has been added. [INFO PR HERE] Port from Horizon. Credits to Azarak.
add: Mining drills, see https://github.com/hrzntal/horizon/pull/276 for more info. Port from Horizon. Credits to Azarak.
/:cl:

### CODE REVIEW

**Comment by Arctaisia** [comment]
Thank you so much for making these Azzy and thank you so much for porting these Gandalf <3

**Comment by Azarak** [comment]
By "your love" he means food. Order me a pizza or something

**Comment by Grunnyyy** [comment]
> By "your love" he means food. Order me a pizza or something

Alright, give me your home address

**Comment by ghost** [comment]
Please sir, just a crumb of screenshots ;-;

**Comment by thestubborn** [comment]
not /vg/ traders smh

**Comment by Gandalf2k15** [comment]
Ready for testing.

File: modular_skyrat/modules/excavation/code/excavation_items.dm [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+/obj/item/strange_rock/attack_self(mob/user)
+	. = ..()
+	if(.)
+		return
+	to_chat(user, "<span class='notice'>You carefully crack open [src].</span>")
+	playsound(src, 'sound/effects/break_stone.ogg', 30, TRUE)
+	for(var/obj/item/I in contents)
+		if(!user.put_in_hand(I))
+			I.forceMove(get_turf(src))
+	qdel(src)
```
Comment (jjpark-kb) [inline modular_skyrat/modules/excavation/code/excavation_items.dm:21]: needs to be changed. xenoarch uses this type I believe

**Review by jjpark-kb — COMMENTED** [review]
False alarm, though I would prefer if it was labeled something else.

**Comment by FlamingLily** [comment]
Looks like looping sounds are runtiming due to components or smth, but i'm not trawling through that thousands of lines of CI logs.

File: modular_skyrat/modules/excavation/code/excavation_items.dm (hunk 1 above)
Comment (Grunnyyy) [reply modular_skyrat/modules/excavation/code/excavation_items.dm:21]: The excavation with strange rocks is the same concept as xenoarch and are basically Azarak Version of it

**Comment by GoldenAlpharex** [comment]
There should be an easy way for ghosts to be able to get the same view as the pilot by clicking onto the console.

**Comment by softcerv** [comment]
is there any way to make shuttles?

**Comment by CandleJaxx** [comment]
god i can't wait until this is fuckin merged

File: code/modules/shuttle/computer.dm [hunk 2]
```diff
@@ -20,17 +20,132 @@
...
 	. = ..()
 	if(!mapload)
 		connect_to_shuttle(SSshuttle.get_containing_shuttle(src))
 
+//SKYRAT EDIT ADDITION
+/obj/machinery/computer/shuttle/attacked_by(obj/item/smacking_object, mob/living/user)
+	if(istype(smacking_object, /obj/item/navigation_datacard))
+		var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttleId)
+		if(M.gateway_stranded)
+			to_chat(user, span_notice("You insert [smacking_object] into [src] loading the navigational data!"))
```
Comment (ghost) [inline code/modules/shuttle/computer.dm]:
```suggestion
			to_chat(user, span_notice("You insert [smacking_object] into [src], loading the navigational data!"))
```
punctuation is cool

Comment (ghost) [reply code/modules/shuttle/computer.dm]: ok upon further review, i found no other typos or issues(that i understood), but have learned that code is scary

**Comment by softcerv** [comment]
This may sound a bit dumb to ask, but do shuttles and stuff work on Icebox?

File: code/controllers/subsystem/shuttle.dm [hunk 3]
```diff
@@ -979,3 +991,11 @@ SUBSYSTEM_DEF(shuttle)
 			has_purchase_shuttle_access |= shuttle_template.who_can_purchase
 
 	return has_purchase_shuttle_access
+
+//SKYRAT EDIT ADDITION
+/datum/controller/subsystem/shuttle/proc/get_transit_instance(atom/movable/movable_atom)
+	for(var/i in transit_instances)
+		var/datum/transit_instance/iterated_transit = i
```
Comment (Kylerace) [inline code/controllers/subsystem/shuttle.dm]: you can use as anything here

File: code/game/objects/items/devices/scanners.dm [hunk 4]
```diff
@@ -660,19 +660,23 @@ GENE SCANNER
...
-
-		for(var/V in SSweather.processing)
-			var/datum/weather/W = V
-			if(W.barometer_predictable && (T.z in W.impacted_z_levels) && W.area_type == user_area.type && !(W.stage == END_STAGE))
-				ongoing_weather = W
-				break
+		//SKYRAT EDIT CHANGE
+		if(weather_controller.current_weathers)
+			for(var/V in weather_controller.current_weathers)
+				var/datum/weather/W = V
```
Comment (Kylerace) [inline code/game/objects/items/devices/scanners.dm]: you can use as anything here

Comment (Gandalf2k15) [reply code/game/objects/items/devices/scanners.dm]: Thanks!

**Comment by Guidesu** [comment]
😢

**Comment by Gandalf2k15** [comment]
Probably not even worth pursuing anymore.

**Comment by CandleJaxx** [comment]
i'm so glad to see this back, it'll be so fucking cool

**Comment by dawsonkeyes** [comment]
yeah but when

**Comment by Guidesu** [comment]
sadness

**Comment by TetrakoHateThis** [comment]
I honestly really really really want this to happen. Space adventures sound like a lot of fun.

**Comment by Paxilmaniac** [comment]
If you'd be interested in using them, I made new versions of the two exploration ships included in this that are still the exact same size, and have the docking port in the same spot:

exploration_crow (the smaller one): 
[here because you can't embed files on github](https://cdn.discordapp.com/attachments/855570952954970125/921972505831419944/exploration_crow_2.dmm)
image

exploration_nexus (the bigger one):
[what if github had file embeds :flushed:](https://cdn.discordapp.com/attachments/855570952954970125/922258915570380861/exploration_nexus_2.dmm)
((ssdecay put dirt and broken tiles all over this for some reason, so just ignore them))
image

IMPORTANT NOTE: I just realized that the version of new nexus that I uploaded had no areas or turf_passthrough, so if you got that before I changed this, I've updated the link to the correct version

**Comment by Gandalf2k15** [comment]
Will reopen once we actually get our shit together lol

**Comment by zeratino** [comment]
sobbing

**Comment by Elli-Skala** [comment]
Why the random baby golith picture?

**Comment by Gandalf2k15** [comment]
> Why the random baby golith picture?

why not?

**Comment by AffectedArc07** [comment]
Please put `[MDB IGNORE]` in your PR title. You are causing excessive system load. 


**EDIT** Thanks dear

**Comment by Arctaisia** [comment]
> Why the random baby golith picture?

Pebbles is INCREDIBLY important... They are the BEST pet..

**Comment by Arctaisia** [comment]
> If you'd be interested in using them, I made new versions of the two exploration ships included in this that are still the exact same size, and have the docking port in the same spot:
> 
> exploration_crow (the smaller one): [here because you can't embed files on github](https://cdn.discordapp.com/attachments/855570952954970125/921972505831419944/exploration_crow_2.dmm) image
> 
> exploration_nexus (the bigger one): [what if github had file embeds 😳](https://cdn.discordapp.com/attachments/855570952954970125/922258915570380861/exploration_nexus_2.dmm) ((ssdecay put dirt and broken tiles all over this for some reason, so just ignore them)) image
> 
> IMPORTANT NOTE: I just realized that the version of new nexus that I uploaded had no areas or turf_passthrough, so if you got that before I changed this, I've updated the link to the correct version

Ooh, these are pretty hot..

**Comment by Paxilmaniac** [comment]
Not sure if it was just my local or not, but when I was messing around with this the ert shuttle couldn't move, as it has no engines

**Comment by Sonic121x** [comment]
- the Ash storm effect are missing,also with the snow storm and sand storm that use the same graphic.
- when playing icebox the outside temputure are too hot for a snow planet.
- lavaland some time have pressure over 50Kpa that make mining equpiment useless against fauna.
- Advanced Shuttles (like the one from here #9951 )that doesn't have a burst engine install can't depart at all.
- when playing blueshift the overmap won't have any space ruin(the big white "!")spawn

(all found from local,not sure it was only happen on my end)

**Comment by Paxilmaniac** [comment]
The biggest issue with ships right now is that their powernets to do not update whenever the shuttle moves, which means that every time the ship moves to a new spot/rotates, every apc inside will stop charging. I believe there was a solution for this made a while ago, but I don't remember exactly?

**Comment by CandleJaxx** [comment]
god i can't wait for this to get merged but Please add the freighter ghostrole ship to the list of Big Pilotable Ships

**Comment by Gandalf2k15** [comment]
https://youtu.be/mXHKjFKBC0g

**Comment by CandleJaxx** [comment]
scarlet-witch-you-took

**Comment by TetrakoHateThis** [comment]
ｉｔ   ｉｓ   ｇｏｎｅ   ａｎｄ   ｗｉｔｈ   ｉｔ   ａｌｌ   ｍｙ   ｈｏｐｅｓ   ａｎｄ   ｄｒｅａｍｓ   ｏｆ   ａ   ｂｅｔｔｅｒ   ｆｕｔｕｒｅ   ｆｏｒ   ｕｓ   ａｌｌ
ａ  ｔｈｒｅａｄ   ｏｆ   ｆａｔｅ   ｈａｓ   ｂｅｅｎ   ｓｅｖｅｒｅｄ   ａｎｄ   ｍａｙ   ｎｅｖｅｒ   ｂｅ   ｂｏｕｎｄ   ａｇａｉｎ
ａｎｄ   ａｌｌ   Ｉ   ａｓｋ   ｉｓ  
ｗｈｙ

**Comment by bruhlookatthisdood** [comment]
oh?
restored?

**Comment by Gandalf2k15** [comment]
i tell you what, this could be good for an event.

**Comment by Erol509** [comment]
Oh, god. Please add it. Well I believe it would make space exploration better along with mining. Overmap is awesome feature, but having it locked only behind mining and research like on aurora would be better, and might prevent station being barebones.

**Comment by Gandalf2k15** [comment]
not today

**Pull request closed without merging**